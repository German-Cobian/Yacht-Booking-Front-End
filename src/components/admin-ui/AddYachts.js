import React, { useState } from 'react';
import Form from 'react-bootstrap/Form';
import Button from 'react-bootstrap/Button';
import axios from 'axios';
import { ToastContainer, toast } from 'react-toastify';
import 'react-toastify/dist/ReactToastify.css';
import { useForm } from 'react-hook-form';
import './admin-ui.css';

function AddYachts() {
  const { reset } = useForm();
  const [yacht, setYacht] = useState({});

  const notifySuccess = () => toast('Yacht added successfully', {
    position: 'top-center',
    autoClose: 15000,
    pauseOnHover: true,
    draggable: true,
  });

  const notifyError = () => toast.error('Error! check error on the logs.', {
    position: 'top-right',
    autoClose: 15000,
    pauseOnHover: true,
    draggable: true,
  });

  const handleChange = (e) => {
    e.preventDefault();
    setYacht({ ...yacht, [e.target.name]: e.target.value });
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    const data = new FormData();
    const img = document.getElementById('image_url');

    data.append('image_url', img.files[0]);
    data.append('name', yacht.name);
    data.append('description', yacht.description);
    data.append('price', yacht.price);

    axios.post('http://127.0.0.1:3001/v1/yachts/', data, {
      headers: {
        'Content-Type': 'multipart/form-data',
      },
    })
      .then(() => {
        notifySuccess();
      })
      .catch(() => {
        notifyError();
      });
    reset();
  };

  return (
    <main className="main">
      <div className="effect" />
      <div className="showcase">
        <Form onSubmit={handleSubmit} className="bg-light rounded-3 shadow p-4 bg-body">
          <Form.Group className="mb-3" controlId="formAddYacht">
            <Form.Label>Yacht Name</Form.Label>
            <Form.Control type="text" htmlFor="name" name="name" placeholder="Enter Yacht Name" onChange={handleChange} />
          </Form.Group>
          <Form.Group className="mb-3" controlId="formAddYacht">
            <Form.Label>Description</Form.Label>
            <Form.Control type="textarea" htmlFor="descripion" name="description" placeholder="Enter Yacht Description" onChange={handleChange} />
          </Form.Group>
          <Form.Group className="mb-3" controlId="formAddYacht">
            <Form.Label>Price</Form.Label>
            <Form.Control type="number" htmlFor="price" name="price" placeholder="Enter Yacht Price" onChange={handleChange} />
          </Form.Group>
          <Form.Group className="mb-3" controlId="formAddYacht">
            <Form.Label>Image</Form.Label>
            <input type="file" id="image_url" name="image_url" placeholder="Select Image" />
          </Form.Group>
          <Button variant="primary" type="submit">
            ADD YACHT
          </Button>
        </Form>
        <ToastContainer />
      </div>
    </main>
  );
}

export default AddYachts;
