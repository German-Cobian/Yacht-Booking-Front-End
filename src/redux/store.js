// import { legacy_createStore as createStore, applyMiddleware } from 'redux';
// import { composeWithDevTools } from 'redux-devtools-extension';
// import thunk from 'redux-thunk';
// import rootReducer from './reducers';
// import fetchYachts from './actions/yachtActions';

// export default createStore(
//   rootReducer,
//   composeWithDevTools(applyMiddleware(thunk)),
// );

// store.dispatch(fetchYachts());
