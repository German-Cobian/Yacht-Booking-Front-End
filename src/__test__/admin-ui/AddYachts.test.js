import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../../redux/storeSetUp';
import AddYachts from '../../components/admin-ui/AddYachts';

it ('renders correctly', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <Router>
        <AddYachts />
      </Router>
    </Provider>
  ).toJSON();
  expect(tree).toMatchSnapshot();
} );