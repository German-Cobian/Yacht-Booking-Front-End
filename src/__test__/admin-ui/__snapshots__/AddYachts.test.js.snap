// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<main
  className="main"
>
  <div
    className="effect"
  />
  <div
    className="showcase"
  >
    <form
      className="bg-light rounded-3 shadow p-4 bg-body"
      onSubmit={[Function]}
    >
      <div
        className="mb-3"
      >
        <label
          className="form-label"
          htmlFor="formAddYacht"
        >
          Yacht Name
        </label>
        <input
          className="form-control"
          htmlFor="name"
          id="formAddYacht"
          name="name"
          onChange={[Function]}
          placeholder="Enter Yacht Name"
          required={true}
          type="text"
        />
      </div>
      <div
        className="mb-3"
      >
        <label
          className="form-label"
          htmlFor="formAddYacht"
        >
          Description
        </label>
        <input
          className="form-control"
          htmlFor="descripion"
          id="formAddYacht"
          name="description"
          onChange={[Function]}
          placeholder="Enter Yacht Description"
          required={true}
          type="textarea"
        />
      </div>
      <div
        className="mb-3"
      >
        <label
          className="form-label"
          htmlFor="formAddYacht"
        >
          Price
        </label>
        <input
          className="form-control"
          htmlFor="price"
          id="formAddYacht"
          name="price"
          onChange={[Function]}
          placeholder="Enter Yacht Price"
          required={true}
          type="number"
        />
      </div>
      <div
        className="mb-3"
      >
        <label
          className="form-label"
          htmlFor="formAddYacht"
        >
          Image
        </label>
        <input
          id="image_url"
          name="image_url"
          placeholder="Select Image"
          required={true}
          type="file"
        />
      </div>
      <button
        className="btn btn-primary"
        disabled={false}
        type="submit"
      >
        ADD YACHT
      </button>
    </form>
    <div
      className="Toastify"
    />
  </div>
</main>
`;
